search/geocoder.cpp

```diff
@@ -312,6 +313,30 @@ size_t OrderCountries(m2::RectD const & pivot, vector<shared_ptr<MwmInfo>> & inf
   auto const sep = stable_partition(infos.begin(), infos.end(), intersects);
   return distance(infos.begin(), sep);
 }
+
+CBV DecimateCianResults(CBV const & cbv)
+{
+  // With the typical amount of buildings in a relevant
+  // mwm nearing 200000, the geocoding slows down considerably.
+  // Leaving only a fraction of them does not seem
+  // to worsen the percieved result.
+  size_t const kMaxCianResults = 10000;
+  static minstd_rand rng;
```

Comment:
Please, init rng with a pre-defined seed. BTW, why not mt19937?

Reply:
No static, please.

Reply:
done

base::RandomSample expects minstd_rand. Probably we may change its signature to accept a UniformRandomBitGenerator, wdyt?
http://en.cppreference.com/w/cpp/concept/UniformRandomBitGenerator